cpython/wasm_assets.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+
+# Adapted from: https://github.com/python/cpython/blob/main/Tools/wasm/wasm_assets.py
+"""Create a WASM asset bundle directory structure.
+The WASM asset bundles are pre-loaded by the final WASM build. The bundle
+contains:
+- a stripped down, pyc-only stdlib zip file, e.g. {PREFIX}/lib/python{version}.zip
+- os.py as marker module {PREFIX}/lib/python3.11/os.py
+- empty lib-dynload directory, to make sure it is copied into the bundle {PREFIX}/lib/python3.11/lib-dynload/.empty
+"""
+
+import argparse
+import pathlib
+import shutil
+import zipfile
+
+# This files are removed from the stdlib by default
+REMOVED_FILES = (
+    # package management
+    "ensurepip/",
+    "venv/",
+    # build system
+    "lib2to3/",
+    # other platforms
+    "_osx_support.py",
+    # Not supported by browser
+    "curses/",
+    "dbm/",
+    "idlelib/",
+    "tkinter/",
+    "turtle.py",
+    "turtledemo",
+    "webbrowser.py",
+)
+
+# This files are unvendored from the stdlib by default
+UNVENDORED_FILES = (
+    "test/",
+    "distutils/",
+    "sqlite3",
+    "ssl.py",
+    "lzma.py",
+)
+
+# TODO: These modules have test directory which we unvendors it separately.
+#       So we should not pack them into the zip file in order to make e.g. import ctypes.test work.
+#       Note that all those tests are moved to the subdirectory of test in Python 3.11,
+#       so we don't need this after Python 3.11 update.
```

Comment:
Maybe just leave it then? It won't be that long before we update to 3.11.

Reply:
If we zip these directories, `loadPyodide('test')` fails because it will try to unzip files into `ctypes/test`, `unittest/test`.

Reply:
> It won't be that long before we update to 3.11.

BTW, are you working on Python 3.11 update recently? If not, I think I can have some time to try.

Reply:
Actually, I just noticed that the upstream commits that move test directories are not applied to Python 3.11, this part is going to be needed until we have Python 3.12...